app/gui2/src/components/widgets/NumericInputWidget.vue

```diff
@@ -82,20 +95,35 @@ const inputStyle = computed<StyleValue>(() => {
   }
 })
 
-function fixupInputValue() {
-  if (inputNode.value != null) inputNode.value.value = `${inputValue.value}`
+function blur() {
+  inputFieldActive.value = false
+  emit('update:modelValue', currentInputValue.value)
+}
+
+/** To prevent other elements from stealing mouse events (which breaks blur),
+ * we instead setup our own `pointerdown` handler while the input is focused.
+ * Any click outside of the input field causes `blur`. */
+function setupAutoBlur() {
+  const options = { capture: true }
+  function callback(event: MouseEvent) {
+    if (blurIfNecessary(inputNode, event)) {
+      window.removeEventListener('pointerdown', callback, options)
+    }
+  }
+  window.addEventListener('pointerdown', callback, options)
 }
 </script>
 
 <template>
   <div class="SliderWidget" v-on="dragPointer.events">
-    <div class="fraction" :style="{ width: sliderWidth }"></div>
+    <div v-if="props.limits != null" class="fraction" :style="{ width: sliderWidth }"></div>
     <input
       ref="inputNode"
-      v-model="inputValue"
+      v-model="currentInputValue"
       class="value"
       :style="inputStyle"
-      @blur="fixupInputValue"
+      @blur="blur"
+      @focus="() => (inputNode && inputNode.select()) || setupAutoBlur()"
```

Comment:
We should set `inputFieldActive` here to true, because the field could be focused by other ways than clicking with mouse. (Then I guess the code setting true in pointer handler would be unnecessary).

Reply:
Done, thanks for suggestion